lib/new-admin/services/login.js

```diff
@@ -1,48 +1,17 @@
-const crypto = require('crypto')
-
 const db = require('../../db')
 
-function generateOTP (name) {
-  const otp = crypto.randomBytes(32).toString('hex')
-
-  const sql = 'insert into one_time_passes (token, name) values ($1, $2)'
-
-  return db.none(sql, [otp, name])
-    .then(() => otp)
-}
-
-function validateOTP (otp) {
-  const sql = `delete from one_time_passes
-    where token=$1
-    returning name, created < now() - interval '1 hour' as expired`
-
-  return db.one(sql, [otp])
-    .then(r => ({ success: !r.expired, expired: r.expired, name: r.name }))
-    .catch(() => ({ success: false, expired: false }))
-}
-
-function register (otp, ua, ip) {
-  return validateOTP(otp)
-    .then(r => {
-      if (!r.success) return r
-
-      const token = crypto.randomBytes(32).toString('hex')
-      const sql = 'insert into user_tokens (token, name, user_agent, ip_address) values ($1, $2, $3, $4)'
+function validateUser (username, password) {
+  const sql = 'SELECT id, username FROM users WHERE username=$1 AND password=$2'
+  const sqlUpdateLastAccessed = 'UPDATE users SET last_accessed = now() WHERE username=$1'
 
-      return db.none(sql, [token, r.name, ua, ip])
-        .then(() => ({ success: true, token: token }))
+  return db.one(sql, [username, password])
+    .then(user => {
+      return db.none(sqlUpdateLastAccessed, [user.username])
+        .then(() => user)
```

Comment:
I think we can either create a tx here or just skip waiting for the last_accessed update.